kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/VaultEdekSerde.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.hashicorp.vault;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+import io.kroxylicious.kms.service.Serde;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+class VaultEdekSerde implements Serde<VaultEdek> {
+
+    @Override
+    public VaultEdek deserialize(@NonNull ByteBuffer buffer) {
+        // TODO use varints once approach is agreed.
+        short kekRefLength = Serde.getUnsignedByte(buffer);
+        if (kekRefLength < 1) {
+            throw new IllegalArgumentException("Unexpected key ref length (%d) read deserializing VaultEdek :".formatted(kekRefLength));
+        }
+        var buf = new byte[kekRefLength];
+        buffer.get(buf);
+        var keyRef = new String(buf, StandardCharsets.UTF_8);
+        int edekLength = buffer.limit() - buffer.position();
+        if (edekLength == 0) {
+            throw new IllegalArgumentException("Zero number of bytes remain in buffer when expecting an edek");
```

Comment:
I don't feel like the message gives any clues as to what the error implies so how about
 ```suggestion
            throw new IllegalArgumentException("unable to deserialize edek as there are zero bytes remaining in the buffer");
```